apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx

```diff
@@ -1,43 +1,153 @@
+      )
+    ) {
+      learningGoalNumber++;
+    }
+
+    return `learningGoal-${learningGoalNumber}`;
+  };
+
+  const emptyKeyConcept = () => {
+    const newKey = generateLearningGoalKey();
+    const newId = newKey;
+    const nextPosition =
+      Math.max(...learningGoalList.map(obj => obj.position)) + 1;
 
-  const renderLearningGoalItems = learningGoalList.map(goal => (
-    <LearningGoalItem
-      deleteItem={() => deleteKeyConcept(goal.id)}
-      key={goal.id}
-    />
-  ));
+    return {
+      key: newKey,
+      id: newId,
+      learningGoal: '',
+      aiEnabled: false,
+      position: nextPosition,
+    };
+  };
+
+  const addNewConceptHandler = event => {
+    event.preventDefault();
+
+    const startingData = emptyKeyConcept();
+
+    const oldLearningGoalList = learningGoalList;
+
+    setLearningGoalList([...oldLearningGoalList, startingData]);
+  };
 
   const deleteKeyConcept = id => {
+    event.preventDefault();
     var updatedLearningGoalList = learningGoalList.filter(
       item => item.id !== id
     );
     setLearningGoalList(updatedLearningGoalList);
   };
 
-  const addNewConceptHandler = event => {
-    // temporary tool for creating unique Ids
-    var updatedId = currentId + 1;
+  const updateLearningGoal = (idToUpdate, keyToUpdate, newValue) => {
+    const newLearningGoalData = learningGoalList.map(learningGoal => {
+      if (idToUpdate === learningGoal.id) {
+        return {
+          ...learningGoal,
+          [keyToUpdate]: newValue,
+        };
+      } else {
+        return learningGoal;
+      }
+    });
+    setLearningGoalList(newLearningGoalData);
+  };
+
+  // TODO: Check that there is at least one programming level here
+  const initialLevelForAssessment = !!rubric ? rubric.levelId : levels[0].id;
+  const [selectedLevelForAssessment, setSelectedLevelForAssessment] = useState(
+    initialLevelForAssessment
+  );
+
+  const saveRubric = event => {
+    event.preventDefault();
+    const dataUrl = !!rubric ? `/rubrics/${rubric.id}/edit` : RUBRIC_PATH;
+    const method = !!rubric ? 'PATCH' : 'POST';
+
+    // Checking that the csrf-token exists since it is disabled on test
+    const csrfToken = document.querySelector('meta[name="csrf-token"]')
+      ? document.querySelector('meta[name="csrf-token"]').attributes['content']
+          .value
+      : null;
+
+    const learningGoalListAsData = transformKeys(learningGoalList);
+
+    const rubric_data = {
+      levelId: selectedLevelForAssessment,
+      lessonId: lessonId,
+      learningGoalsAttributes: learningGoalListAsData,
+    };
 
-    setCurrentId(updatedId);
+    fetch(dataUrl, {
+      method: method,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-CSRF-Token': csrfToken,
+      },
+      body: JSON.stringify(rubric_data),
+    })
+      .then(response => response.json())
+      .then(data => {
+        navigateToHref(data.redirectUrl);
```

Comment:
just noting that you'll probably want to only do this when creating a rubric, rather than on update. This doesn't need to be resolved in this PR though